### DISCUSSION TITLE
Can I make EditorWindow with this plugins?

### DISCUSSION THREAD

**Discussion by locus84** [TylerTemp/SaintsField#109] [Q&A] [answered]
Thank you for great plugin

NaughtyAttributes has EditorGUILayout.PropertyField replacement.
Is there something like that? or should I use different approach?

**Comment by TylerTemp** [comment, marked as answer]
That's a in plan feature (but not documented in milestone yet). However, even do, I'd prefer to implement one like Odin's `OdinWindow`, rather than a thing like NaughtyAttribute's solution.

At the moment it's not possible, yet.

-----------------------------

Before this feature is added, this is what I do daily on my work (only example of UI Toolkit, won't work for IMGUI):

```csharp
public class InRoundStatisticsPanel: ScriptableObject
{
    public string[] files;

    [Button]
    public void ClickButton()
    {
        Debug.Log("clicked");
    }
}

public class EditorInRoundStatisticsPanelEditorWindow: EditorWindow
{
    [MenuItem("Tools/MyTools")]
    public static void OpenWindow()
    {
        EditorWindow window = GetWindow(typeof(EditorInRoundStatisticsPanelEditorWindow), false, "MyTitle");
        window.Show();
    }

    private InRoundStatisticsPanel _target;

    public void CreateGUI()
    {
        VisualElement root = rootVisualElement;
        _target ??= CreateInstance<InRoundStatisticsPanel>();  // or load data from some source
        // do some data populate here...

        UnityEditor.Editor editor = UnityEditor.Editor.CreateEditor(_target, typeof(SaintsEditor));
        root.Add(new InspectorElement(editor));
    }

    private void OnDisable()
    {
        // TODO: save your changes of _target here, if you want
    }
}
```

**Reply by locus84** [reply]
Looks like a valid alternative solution, the SO does not need serialization so it may be used like a helper class. or I can place it inner class of EditorWindowClass

**Reply by TylerTemp** [reply]
> Looks like a valid alternative solution, the SO does not need serialization so it may be used like a helper class. or I can place it inner class of EditorWindowClass

Yes, you can inline it inside the EditorWindowClass, and even making it private is OK

**Comment by TylerTemp** [comment]
Hi,

This feature has been added in [3.12.0](https://github.com/TylerTemp/SaintsField/releases/tag/3.12.0)

Now you can just do:

```csharp
using SaintsField.Editor;

public class Example: SaintsEditorWindow
{
    [MenuItem("Tools/Example")]
    public static void OpenWindow()
    {
        EditorWindow window = GetWindow<Example>(false, "Scriptable Editor");
        window.Show();
    }

    // your logic here
    public string[] files;

    [Button]
    public void ClickButton()
    {
        Debug.Log("clicked");
    }
}
```

And also two showcase, already documented in the README and [saintsfield.comes.today](https://saintsfield.comes.today/saintseditorwindow/usage--example)

[![video](https://github.com/user-attachments/assets/a724f450-0d36-4b39-aada-c180d2d8990b)](https://github.com/user-attachments/assets/f80f4de2-4d12-47dc-be82-aff2e3ba2c0b)

[![video](https://github.com/user-attachments/assets/855b628a-0c3a-4cc1-90b7-115d401f1cfa)](https://github.com/user-attachments/assets/665ebe8a-22b1-4df8-8f20-47fa8b23e3b9)
